WcaOnRails/app/models/user.rb

```diff
@@ -496,35 +496,36 @@ def editable_fields_of_user(user)
       return fields
     end
     if user == self
-      fields << :current_password
-      fields << :password << :password_confirmation
-      fields << :email
-      fields << :preferred_events
+      fields.merge %i(
+        current_password password password_confirmation
+        email preferred_events results_notifications_enabled
+      )
       fields << { user_preferred_events_attributes: [:id, :event_id, :_destroy] }
-      fields << :results_notifications_enabled
     end
     if admin? || board_member?
-      fields << :delegate_status
-      fields << :senior_delegate_id
-      fields << :region
+      fields.merge %i(delegate_status senior_delegate_id region)
+    end
+    if user.any_kind_of_delegate? && (user == self || user.senior_delegate == self || admin? || board_member)
+      fields.merge %i(location_description phone_number notes)
     end
     if admin? || any_kind_of_delegate?
-      fields << :wca_id << :unconfirmed_wca_id
-      fields << :avatar << :avatar_cache
+      fields.merge %i(
+        wca_id unconfirmed_wca_id
+        avatar avatar_cache
+      )
     end
     if user == self || admin? || any_kind_of_delegate?
       cannot_edit_data = !!cannot_edit_data_reason_html(user)
       if !cannot_edit_data
-        fields << :name
-        fields << :dob
-        fields << :gender
-        fields << :country_iso2
+        fields.merge %i(name dob gender country_iso2)
       end
       fields += CLAIM_WCA_ID_PARAMS
```

Comment:
Ugh... When I tested it in my console, I mistyped... Sorry!

Yeah, I like `+=` a lot more than`.merge`!